static_frame/test/property/test_util.py

```diff
@@ -306,6 +306,14 @@ def test_intersect1d(self, arrays: tp.Sequence[np.ndarray]) -> None:
             return
         post = util.intersect1d(arrays[0], arrays[1], assume_unique=False)
         self.assertTrue(post.ndim == 1)
+
+        # Coerce NaTs to Nones to ensure set operations work as expected
+        if resolve_dtype(*(arr.dtype for arr in arrays)).kind == 'O':
+            for i in range(len(arrays)):
+                arr = arrays[i]
+                if arr.dtype.kind in 'Mm' and np.isnat(arr).any():
+                    arrays[i] = arrays[i].astype(object)
+
```

Comment:
This was caught by hypothesis, but it is unrelated to my PR, as it also fails on default (as of d561ee05d686269507ca160a314c35b908d11232)

The falsifying example was `@reproduce_failure('6.92.1', b'AAAFAAIBAAABAIAAAAAAAAAAAgEA')`

`arrays` is `[array(['NaT'], dtype='datetime64[D]'), array([None], dtype=object)]`